content/5_Plat/Geo_Pri.mdx

```diff
@@ -1,14 +1,213 @@
 ---
 id: geo-pri
 title: 'Geometry Primitives'
-author: Benjamin Qi
+author: Mihnea Brebenel, Benjamin Qi
 description: Basic setup for geometry problems.
 frequency: 1
 ---
 
 You should know basic operations like cross product and dot product.
 
-## Tutorial
+## Standard Problems
+
+<FocusProblem problem="pointlocationtest" />
+
+To check the $P$ location towards the $P_1$ $P_2$ line we use following formula: $(P.y - P_1.y) * (P_2.x - P_1.x) - (P.x - P_1.x) * (P_2.y - P_1.y)$
+If it's equal to $0$ it means that $P$, $P_1$ and $P_2$ are collinear. Otherwise the value's sign indicated if $P$ is under the line - negative - or above the line - positive.
+
+<Spoiler title="Demonstration">
+
+Let $m_1=\frac{P_1.y - P_2.y}{P_1.x - P_2.x}$ be the slope of the linear function that goes through the points $P_1$ and $P_2$ and
+$m_2=\frac{P_2.y - P.y}{P_2.x - P.x}$ be the slope of the linear function that goes through the points $P_2$ and $P$.
+
+The points are collinear if and only if the slopes are equal i.e. $m_1 = m_2 \Leftrightarrow \frac{P_1.y - P_2.y}{P_1.x - P_2.x} = \frac{P_2.y - P.y}{P_2.x - P.x}
+\Leftrightarrow (P_1.y - P_2.y) * (P_2.x - P.x) = (P_1.x - P_2.x) * (P_1.x - P_2.x) \ Leftrightarrow (P_1.y - P_2.y) * (P_2.x - P.x) - (P_1.x - P_2.x) * (P_1.x - P_2.x) = 0$.
+And this way we obtained the previously mentioned formula.
+
+This formula doesn't only tell us whether the points are collinear, but where the point $P$ is located too.
+
+![Dem](/assets/dem-collinear.png)
+
+</Spoiler>
+
+### Implementation
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+// BeginCodeSnip{Point Class}
+struct Point {
+	int x, y;
+	Point(int a = 0, int b = 0) : x(a), y(b) {}
+
+	friend istream &operator>>(istream &in, Point &p) {
```

Comment:
is `friend` needed? all the members are public anyways

Reply:
It gives an error without friend

Reply:
i think that's cause you have to move it out of the class itself, but whatever

Reply:
Yep, this way works. Do you want it outside or inside?

Reply:
now that i think about it it doesn't matter, keep it as it is